python/cudf/cudf/core/indexed_frame.py

```diff
@@ -1800,6 +1800,60 @@ def drop_duplicates(
             self._index.names if not ignore_index else None,
         )
 
+    @_cudf_nvtx_annotate
+    def duplicated(self, subset=None, keep="first"):
+        """
+        Return boolean Series denoting duplicate rows.
+
+        Considering certain columns is optional.
+
+        Parameters
+        ----------
+        subset : column label or sequence of labels, optional
+            Only consider certain columns for identifying duplicates, by
+            default use all of the columns.
+        keep : {'first', 'last', False}, default 'first'
+            Determines which duplicates (if any) to mark.
+
+            - ``first`` : Mark duplicates as ``True`` except for the first occurrence.
+            - ``last`` : Mark duplicates as ``True`` except for the last occurrence.
+            - False : Mark all duplicates as ``True``.
+        """  # noqa: E501
+        if subset is None:
+            subset = self._column_names
+        elif (
+            not np.iterable(subset)
+            or isinstance(subset, str)
+            or isinstance(subset, tuple)
+            and subset in self._data.names
+        ):
+            subset = (subset,)
+        diff = set(subset) - set(self._data)
+        if len(diff) != 0:
+            raise KeyError(f"columns {diff} do not exist")
+
+        if isinstance(self, cudf.Series):
+            df = self.to_frame(name="None")
+            subset = ["None"]
+        else:
+            df = self.copy(deep=False)
+        df._data["index"] = cudf.core.column.arange(
```

Comment:
Should this use RangeIndex somehow? I hesitate to see hardcoded `size_type`.

Reply:
By `RangeIndex` did you intend to do `df._data["index"] = cudf.RangeIndex(0, x)._as_int_index()`? This by the way still returns an `int64` dtype rather than `int32`.

Reply:
I would have expected there to be a better way to generate a sequence of indices. We hardcoded `"int32"` below, which I assume is only because of the definition of `using size_type = int32_t;` in C++. If possible we should use some other way to generate this sequence of indices that is already aware of the libcudf index type and doesn't require hardcoding `"int32"`. We're trying to prepare for a future where we could potentially replace `int32` indices with `int64` types with minimal code changes.

Reply:
Should we have some kind of a mapping in cython to `size_type` <-> `np.dtype`?

Reply:
We already have that partially done here: https://github.com/rapidsai/cudf/blob/8e0ee53fea2db1e5d8a7725049756237e7bf9438/python/cudf/cudf/_lib/cpp/types.pxd#L7

Maybe we need a `size_type_dtype` or similar.

Reply:
Note: there's a similar definition using Numba types in `strings_udf`, too:
https://github.com/rapidsai/cudf/blob/8e0ee53fea2db1e5d8a7725049756237e7bf9438/python/strings_udf/strings_udf/_typing.py#L17-L18

Reply:
Added a `size_type_dtype` to `cudf._typing` to be in line with `strings_udf`. There are soo many places that I see a candidate fit to use this variable, but I'm refraining from polluting this PR with those changes, will open a separate PR.

Reply:
I think this should be in `cudf._lib.types` instead. The `_typing` module is more for defining types for static analysis (mypy)

Reply:
> I think this should be in `cudf._lib.types` instead. The `_typing` module is more for defining types for static analysis (mypy)

Done 👍